src/client/pfs.go

```diff
@@ -410,43 +388,58 @@ func (c APIClient) FlushCommit(commits []*pfs.Commit, toRepos []*pfs.Repo, cb fu
 	}
 }
 
-// FlushCommitAll returns commits that have the specified `commits` as
-// provenance. Note that it can block if jobs have not successfully
-// completed. This in effect waits for all of the jobs that are triggered by a
-// set of commits to complete.
-//
-// If toRepos is not nil then only the commits up to and including those repos
-// will be considered, otherwise all repos are considered.
-//
-// Note that it's never necessary to call FlushCommit to run jobs, they'll run
-// no matter what, FlushCommitAll just allows you to wait for them to complete and
-// see their output once they do.
-func (c APIClient) FlushCommitAll(commits []*pfs.Commit, toRepos []*pfs.Repo) ([]*pfs.CommitInfo, error) {
-	var cis []*pfs.CommitInfo
-	if err := c.FlushCommit(commits, toRepos, func(ci *pfs.CommitInfo) error {
-		cis = append(cis, ci)
+// InspectCommitset returns info about a specific Commitset.
+func (c APIClient) InspectCommitset(id string) ([]*pfs.CommitInfo, error) {
+	result := []*pfs.CommitInfo{}
+	if err := c.inspectCommitset(id, false, func(ci *pfs.CommitInfo) error {
+		result = append(result, ci)
 		return nil
 	}); err != nil {
 		return nil, err
 	}
-	if len(cis) > 0 {
-		sort.Slice(cis, func(i, j int) bool {
-			return len(cis[i].Provenance) < len(cis[j].Provenance)
-		})
+	return result, nil
+}
+
+// BlockCommitset blocks until all of a Commitset's commits are finished.  To
+// wait for an individual commit, use BlockCommit instead.
+func (c APIClient) BlockCommitsetAll(id string) ([]*pfs.CommitInfo, error) {
```

Comment:
It would probably be better to have `BlockCommitsetAll` call `BlockCommitset` rather than inspect commit. That would be a safer pattern for ensuring the same semantics. Also, you can just return the `inspectCommitset` call in `BlockCommitset`.

Reply:
Also, in regards to the gRPC scrubbing, it would probably be safer to handle it right at the API boundary (set up the defer immediately in an exported method) rather than in the internal methods. It seems like it would be easier to ensure that we are doing it.